demux/demux_mkv.c

```diff
@@ -390,6 +390,10 @@ static bstr demux_mkv_decode(struct mp_log *log, mkv_track_t *track,
             }
             size = dstlen - out_avail;
         } else if (enc->comp_algo == 3) {
+            if (enc->comp_settings_len == 0 || !enc->comp_settings) {
```

Comment:
in this function also bad things happen if `!HAVE_ZLIB` and a track is zlib-compressed

Reply:
I don't think it is that bad, it will just skip all the zlib compressed elements.